applications/StructuralMechanicsApplication/custom_elements/updated_lagrangian.cpp

```diff
@@ -361,6 +361,25 @@ double UpdatedLagrangian::CalculateDerivativesOnReferenceConfiguration(
 
     J0 = this->GetGeometry().Jacobian( J0, PointNumber, ThisIntegrationMethod, delta_displacement);
 
+    const SizeType working_space_dimension = GetGeometry().WorkingSpaceDimension();
+    const SizeType local_space_dimension = GetGeometry().LocalSpaceDimension();
+    if (J0.size1() != working_space_dimension || J0.size2() != local_space_dimension)
+        J0.resize(working_space_dimension, local_space_dimension, false);
+
+    const Matrix& r_shape_functions_gradient_in_integration_point = GetGeometry().ShapeFunctionsLocalGradients(ThisIntegrationMethod)[PointNumber];
+
+    J0.clear();
```

Comment:
it is not the update, it is the different way to obtain the reference pseudo initial Jacobian.

This way is obviously only valid for MPM-wise reset elements.

Reply:
I don't understand, what are you trying to do, moving the mesh? Should you re-initialize the element instead?

Reply:
In MPM I reset the mesh after every step, including displacements. Thus, the displacements are zero and the computation with the offset won't work properly.

Reply:
Let's see. 
![imagen](https://user-images.githubusercontent.com/19991680/107926734-56eb0900-6f76-11eb-91da-f0646d11c77a.png)

This is the base of the MPM method. In here the solid UL corresponds with the background mesh or the solid UL elements are independent?, because if that is the case, can you just move the relative to MPM elements and keep the solid UL as independent?

Reply:
I basically do step 1 3 and 4 in the process. Step 2 is done in the UL element.

Now main design issue is, not using the MPM application and it's elements at all. 

Basically in step 2 the mpm particle or element is dependent on the mesh. But it doesn't has DISPLACEMENTS to compute the Jacobian at the delta position nor a movement of the mesh at all.

Reply:
One possible solution would be to derive an element which just overrides this function. I reckon that would resolve the problem. But maybe there's a note elegant solution.

Reply:
OK, let's say that what you actually want is to keep the mesh static (do not move it), but the initial deformation gradients must be preserved for a consistency.

Reply:
I think i need to check the code to see how to do it.